src/python/WMCore/WMRuntime/Monitors/PerformanceMonitor.py

```diff
@@ -182,33 +183,47 @@ def periodicUpdate(self):
             # Then we have no step PID, we can do nothing
             return
 
-        # Now we run the monitor command and collate the data
-        cmd = self.monitorBase % (stepPID, stepPID)
-        stdout, _stderr, _retcode = subprocessAlgos.runCommand(cmd)
+        # Now we run the ps monitor command and collate the data
+        # Gathers RSS, %CPU and %MEM statistics from ps
+        ps_cmd = self.monitorBase % (stepPID, stepPID)
+        stdout, _stderr, _retcode = subprocessAlgos.runCommand(ps_cmd)
 
-        output = stdout.split()
-        if not len(output) > 6:
+        ps_output = stdout.split()
+        if not len(ps_output) > 6:
             # Then something went wrong in getting the ps data
             msg = "Error when grabbing output from process ps\n"
-            msg += "output = %s\n" % output
-            msg += "command = %s\n" % self.monitorCommand
+            msg += "output = %s\n" % ps_output
+            msg += "command = %s\n" % ps_cmd
             logging.error(msg)
             return
 
-        # ps returns data in kiloBytes, let's make it megaBytes
-        # I'm so confused with these megabytes and mebibytes...
-        rss = int(output[2]) // 1000  # convert it to MiB
-        logging.info("Retrieved following performance figures:")
-        logging.info("RSS: %s; PCPU: %s; PMEM: %s", output[2], output[3], output[4])
+        # run the command to gather PSS memory statistics from /proc/<pid>/smaps
+        smaps_cmd = self.pssMemoryCommand % (stepPID)
+        stdout, _stderr, _retcode = subprocessAlgos.runCommand(smaps_cmd)
+
+        smaps_output = stdout.split()
+        if not len(smaps_output) == 1:
+            # Then something went wrong in getting the smaps data
+            msg = "Error when grabbing output from smaps\n"
+            msg += "output = %s\n" % smaps_output
+            msg += "command = %s\n" % smaps_cmd
+            logging.error(msg)
+            return
+
+        # smaps also returns data in kiloBytes, let's make it megaBytes
+        # I'm also confused with these megabytes and mebibytes...
+        pss = int(smaps_output[0]) // 1000
+
+        logging.info("PSS: %s; RSS: %s; PCPU: %s; PMEM: %s", smaps_output[0], ps_output[2], ps_output[3], ps_output[4])
 
         msg = 'Error in CMSSW step %s\n' % self.currentStepName
         msg += 'Number of Cores: %s\n' % self.numOfCores
 
-        if self.maxRSS is not None and rss >= self.maxRSS:
-            msg += "Job has exceeded maxRSS: %s\n" % self.maxRSS
-            msg += "Job has RSS: %s\n" % rss
+        if self.maxPSS is not None and pss >= self.maxPSS:
+            msg += "Job has exceeded maxPSS: %s\n" % self.maxPSS
```

Comment:
@goughes sorry for suggesting it only now, but can you make another small modification here such that the error message is more clear? I'd suggest to add the unit in the end of these 2 lines, e.g.
`msg += "Job has exceeded maxPSS: %s MB\n" % self.maxPSS`

Don't forget to squash the new commit against the first one ;)